### DISCUSSION TITLE
Why not use FetchContent_MakeAvailable()?

### DISCUSSION THREAD

**Discussion by melroy89** [nlohmann/json#3345] [Ideas]
Your example shows regarding CMake FetchCOntent: https://github.com/nlohmann/json#embedded-fetchcontent

```
FetchContent_Declare(json
  URL https://github.com/nlohmann/json/releases/download/v3.10.5/json.tar.xz
)
FetchContent_GetProperties(json)
if(NOT json_POPULATED)
  FetchContent_Populate(json)
  add_subdirectory(${json_SOURCE_DIR} ${json_BINARY_DIR} EXCLUDE_FROM_ALL)
endif()
```

This should be avoid according to the official CMake docs: https://cmake.org/cmake/help/git-stage/module/FetchContent.html#id2

Prefer to use `FetchContent_MakeAvailable()` instead of custom logic:

```cmake
FetchContent_Declare(json URL https://github.com/nlohmann/json/releases/download/v3.10.5/json.tar.xz)
FetchContent_MakeAvailable(json)
```

Regards, Melroy

**Comment by nlohmann** [comment]
Thanks for the hint - I was not aware of this. Maybe (?) the verbose version was used as it requires an earlier CMake version. I will change the documentation.

**Reply by nlohmann** [reply]
See https://github.com/nlohmann/json/pull/3351

**Reply by nlohmann** [reply]
Updated: https://json.nlohmann.me/integration/cmake/#fetchcontent
